File: dot-net/src/ISP.Animal/ISP.Animal/Model/Perro.cs
namespace ISP.Animal.Model
{
    public class Perro : Animal
    {
        private bool _alimentado;
        private bool _acariciado;

        public override void Alimentar()
        {
            _alimentado = true;
        }

        public override void Acariciar()
        {
            _acariciado = true;
        }

        public override string GetEstado()
        {
            var estado = string.Empty;

            if (_alimentado)
				estado = "no quiero un hueso";
            if (!_alimentado)
				estado = "quiero un hueso";
            if (!_acariciado)
				estado += " y quiero jugar";

            return estado;
        }
    }
}

File: java/isp-animal/src/Perro.java

public class Perro extends Animal {

    private boolean alimentado = false;
    private boolean acariciado = false;

    @Override
	public void alimentar() {
    	this.alimentado = true;
	}

	@Override
	public void acariciar() {
		this.acariciado = true;
	}

	@Override
	public String getEstado() {
        String estado = "";

        if (this.alimentado)
			estado = "no quiero un hueso";
        if (!this.alimentado)
			estado = "quiero un hueso";
        if (!this.acariciado)
			estado += " y quiero jugar";

        return estado;
	}
}
